### PR TITLE
Partly implement XML child and elements

### PR DESCRIPTION
**Pull request by evilpie**
This makes #3294 (rollercoaster-creator-2) fully playable.

Missing is any (*) matching for child()/elements() and the existing attributes() method. Also missing is support for number indexes with child().

### CODE REVIEW

File: core/src/avm2/globals/xml.rs [hunk 1]
```diff
@@ -88,6 +88,49 @@ pub fn to_xml_string<'gc>(
...
+}
+
+pub fn child<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let xml = this.unwrap().as_xml_object().unwrap();
+    let multiname = name_to_multiname(activation, &args[0])?;
+    let children = if let E4XNodeKind::Element { children, .. } = &*xml.node().kind() {
```
Comment (Aaron1011) [inline core/src/avm2/globals/xml.rs:121]: Could you add a FIXME here about checking when the name is an index?

File: tests/tests/swfs/avm2/xml_child/Test.as [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package {
+  import flash.display.Sprite;
+  public class Test extends Sprite { }
+}
+
+var xml = new XML("<x><foo>foo1</foo><bar>bar1</bar><foo>foo2</foo></x>")
```
Comment (Aaron1011) [inline tests/tests/swfs/avm2/xml_child/Test.as:6]: Could you also test `elements` in this file, or in a new test?

File: core/src/avm2/globals/xml.rs [hunk 3]
```diff
@@ -88,6 +88,50 @@ pub fn to_xml_string<'gc>(
...
+}
+
+pub fn child<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let xml = this.unwrap().as_xml_object().unwrap();
+    let multiname = name_to_multiname(activation, &args[0])?;
+    // FIXME: Support numerical indexes.
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/xml.rs:121]: I think this should also mention calling `XML.child` with any ("*") as the parameter

**Review by Aaron1011 — APPROVED** [review]

File: core/src/avm2/globals/xml.rs (hunk 3 above)
Comment (Aaron1011) [reply core/src/avm2/globals/xml.rs:121]: This is being addressed in a separate PR

**Pull request merged**